### PR TITLE
Bugfix: Image not found error causes a very long and uncontrolled stall

### PR DESCRIPTION
**Pull request by erdemtoraman**
#153

1. Pass the context to backoff strategy so that if the ctx is cancelled it doesn't proceed
2.  If the error is a not found error, treat it as a permanent one so that it's not retried and immediately returned

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,36 +1,17 @@
-module github.com/testcontainers/testcontainers-go
+module github.com/erdemtoraman/testcontainers-go
```
Comment (gianarb) [inline go.mod]: Hello!
The change I see in go.mod does not look right :) can you double-check?

**Comment by gianarb** [comment]
@erdemtoraman the difference in go.mod go.sum does not look right, you didn't add any dependency in your code, so my expectation is for those to be untouched. I do not know why they are differnet

**Comment by erdemtoraman** [comment]
> @erdemtoraman the difference in go.mod go.sum does not look right, you didn't add any dependency in your code, so my expectation is for those to be untouched. I do not know why they are differnet

Yeah you're right, I have no idea what might have been there, i'll check out

**Comment by gianarb** [comment]
I am sorry if I look to picky @erdemtoraman but if you look at the commits you pushed on this branch it is clear that something is not right

https://github.com/testcontainers/testcontainers-go/pull/155/commits

You probably started your branch from the wrong point.
Do you mind to figure out why it looks like there is a merge commit https://github.com/testcontainers/testcontainers-go/pull/155/commits/ea1f33afc2be544b267dfaf5c0aff44ef264abc7 and things like that?
Take your time and if you need some help with cherry-picking commits I can help

**Comment by Bablzz** [comment]
It seems to me, in go.sum remove new line (maybe IDE delete new line as default)

**Comment by ClaytonNorthey92** [comment]
just one suggestion, could we squash this into two commits?
* travis version update
* the backoff fix

**Comment by gianarb** [comment]
I cherry-picked a bunch of your commits into this PR https://github.com/testcontainers/testcontainers-go/pull/162

Thanks!

**Comment by erdemtoraman** [comment]
> I cherry-picked a bunch of your commits into this PR #162
> 
> Thanks!

Yeah sorry for being so slow with this, thanks!

**Pull request closed without merging**